contracts/axelarnet-gateway/src/contract/execute.rs

```diff
@@ -0,0 +1,134 @@
+use axelar_wasm_std::nonempty;
+use cosmwasm_std::{to_json_binary, Addr, Api, HexBinary, Response, Storage, WasmMsg};
+use error_stack::{report, Result, ResultExt};
+use router_api::client::Router;
+use router_api::{Address, ChainName, CrossChainId, Message};
+use sha3::{Digest, Keccak256};
+
+use crate::contract::Error;
+use crate::events::AxelarnetGatewayEvent;
+use crate::msg::AxelarExecutableExecuteMsg;
+use crate::state::{self};
+
+// TODO: Retrieve the actual tx hash from core, since cosmwasm doesn't provide it. Use a placeholder in the meantime.
+const PLACEHOLDER_TX_HASH: &str =
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
```

Comment:
Idk how I feel about this. What's the purpose of adding placeholder tx hash? Why not just use a counter by itself, or blockhash and counter?

Reply:
it'll be switched to the actual tx hash, so just a placeholder for now

Reply:
I'd rather use the blockhash as a placeholder for now, instead of this dummy string. The blockhash provides more info than this.

Reply:
Sure, blockhash isn't available, but added block height